src/main/java/org/elasticsearch/index/mapper/internal/TimestampFieldMapper.java

```diff
@@ -283,7 +282,14 @@ public void merge(Mapper mergeWith, MergeContext mergeContext) throws MergeMappi
                 this.enabledState = timestampFieldMapperMergeWith.enabledState;
             }
         } else {
-            if (!timestampFieldMapperMergeWith.defaultTimestamp().equals(defaultTimestamp)) {
+            if (timestampFieldMapperMergeWith.defaultTimestamp() == null && defaultTimestamp == null) {
+                return;
+            }
+            if (timestampFieldMapperMergeWith.defaultTimestamp() != null && defaultTimestamp == null) {
```

Comment:
I think you can just do the `== null` part of this and the next `if` condition, since if the other were equal to null, we would have already returned in the first `if`?

```
if (x == null && y == null) return
if (x == null) ...
else if (y == null) ...
else // x and y are not null
```